src/Microsoft.HttpRepl.Tests/Suggestions/HeaderCompletionTests.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net.Http;
+using Microsoft.HttpRepl.Fakes;
+using Microsoft.HttpRepl.FileSystem;
+using Microsoft.HttpRepl.Preferences;
+using Microsoft.HttpRepl.Suggestions;
+using Microsoft.HttpRepl.UserProfile;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.Suggestions
+{
+    public class HeaderCompletionTests
+    {
+        [Fact]
+        public void GetCompletions_NullExistingHeaders_ProperResults()
+        {
+            IEnumerable<string> result = HeaderCompletion.GetCompletions(null, "U");
+
+            Assert.Equal(3, result.Count());
+            Assert.Contains("Upgrade", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("Upgrade-Insecure-Requests", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("User-Agent", result, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void GetCompletions_ExistingHeader_ProperResults()
+        {
+            IReadOnlyCollection<string> existingHeaders = new Collection<string>() { "User-Agent" };
+
+            IEnumerable<string> result = HeaderCompletion.GetCompletions(existingHeaders, "U");
+
+            Assert.Equal(2, result.Count());
+            Assert.Contains("Upgrade", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("Upgrade-Insecure-Requests", result, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void GetCompletions_NotAnExistingHeader_ProperResults()
+        {
+            IReadOnlyCollection<string> existingHeaders = new Collection<string>() { "Content-Type" };
+
+            IEnumerable<string> result = HeaderCompletion.GetCompletions(existingHeaders, "U");
+
+            Assert.Equal(3, result.Count());
+            Assert.Contains("Upgrade", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("Upgrade-Insecure-Requests", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("User-Agent", result, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void GetCompletions_NoResults_ReturnsNull()
+        {
+            IEnumerable<string> result = HeaderCompletion.GetCompletions(null, "Not-A-Real-Header");
+
+            Assert.Null(result);
```

Comment:
spec question: would we rather return (and therefore have to be aware of) null, or would it be better to return an empty collection?

Reply:
A quick glance at the code makes it seem like returning an empty collection would be fine as well. The code in both places where this is used is written to handle both scenarios. I think we could simplify it to return an empty collection now that we have the tests to be able to assert that it doesn't return null. Let me take a little deeper look and I'll make a call.

Reply:
OK, I changed that up in 425e081.